app/src/main/res/layout/nav_header.xml

```diff
@@ -13,17 +14,33 @@
     android:padding="15dp"
     android:theme="@style/ThemeOverlay.AppCompat.Dark">
 
-    <de.hdodenhof.circleimageview.CircleImageView
-        android:id="@+id/nav_header_imageView"
-        android:layout_width="55dp"
-        android:layout_height="55dp"
-        android:src="@drawable/icon_nav"
-        app:civ_border_color="@color/white"
-        app:civ_border_width="1dp"
-        android:layout_gravity="top|start"
-        android:layout_marginStart="5dp"
-        android:layout_marginTop="5dp"
-        />
+    <RelativeLayout
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content">
+
+        <de.hdodenhof.circleimageview.CircleImageView
+            android:id="@+id/nav_header_imageView"
+            android:layout_width="55dp"
+            android:layout_height="55dp"
+            tools:src="@drawable/icon_nav"
+            app:civ_border_color="@color/white"
+            app:civ_border_width="1dp"
+            android:layout_gravity="top|start"
+            android:layout_marginStart="5dp"
+            android:layout_marginTop="5dp"
+            />
+
+        <ImageView
+            android:id="@+id/nav_header_letterView"
+            android:layout_width="55dp"
+            android:layout_height="55dp"
+            tools:src="@drawable/icon_nav"
+            android:layout_gravity="top|start"
+            android:layout_marginStart="5dp"
+            android:layout_marginTop="5dp"
```

Comment:
Initially make the visibility gone. Make it visible only when user signs in using email & pass and profile pic is null. 

Even in the case of Google sign in, when profile pic is not present, the app is using the character as if it is the image and displayed in circularImageView which is fine.